### PR TITLE
feat: support delete cluster from UI

### PR DESCRIPTION
**Pull request by haoshuwei**
fix issue [#3551](https://github.com/argoproj/argo-cd/issues/3551)
Signed-off-by: haoshuwei <haoshuwei24@gmail.com>

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [] Optional. My organization is added to USERS.md.
* [X] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by haoshuwei** [comment]
The picture below shows the feat of delete cluster from ui
image

File: ui/src/app/settings/components/clusters-list/clusters-list.tsx [hunk 1]
```diff
@@ -1,49 +1,79 @@
...
-                                        <div className='row'>
-                                            <div className='columns small-3'>
-                                                <i className='icon argo-icon-hosts' /> {clusterName(cluster.name)}
-                                            </div>
-                                            <div className='columns small-5'>{cluster.server}</div>
-                                            <div className='columns small-2'>{cluster.serverVersion}</div>
-                                            <div className='columns small-2'>
-                                                <ConnectionStateIcon state={cluster.connectionState} /> {cluster.connectionState.status}
+export const ClustersList = (props: RouteComponentProps<{}>) => {
+    const [refreshToken, setRefreshToken] = React.useState(0);
```
Comment (alexmt) [inline ui/src/app/settings/components/clusters-list/clusters-list.tsx:10]: Please use data loader reload method instead, please. The account-details.tsx has an example:

https://github.com/argoproj/argo-cd/blob/301d18820a2bb2648125c30c8b20068553738eaa/ui/src/app/settings/components/account-details/account-details.tsx#L59

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you for contribution @haoshuwei ! Added one minor comment.

File: ui/src/app/settings/components/clusters-list/clusters-list.tsx (hunk 1 above)
Comment (haoshuwei) [reply ui/src/app/settings/components/clusters-list/clusters-list.tsx]: Thanks for your comments,  I have updated it.

**Comment by haoshuwei** [comment]
@alexmt Thanks for your comments, I have updated it and all checks have passed.

**Review by alexmt — APPROVED** [review]
LGTM!

**Pull request merged**